src/ui.navigation-progress/composables/attrs.composable.js

```diff
@@ -4,45 +4,42 @@ import { cva, cx } from "../../service.ui";
 import defaultConfig from "../configs/default.config";
 import { computed } from "vue";
 
-export function useAttrs(props) {
+export default function useAttrs(props) {
   const { config, getAttrs, getColor, setColor, isSystemKey } = useUI(
     defaultConfig,
     () => props.config,
   );
+
   const attrs = {};
 
   for (const key in defaultConfig) {
     if (isSystemKey(key)) continue;
 
-    let classes = "";
-    let value = config.value[key];
-
-    const isVariants = value.variants || value.compoundVariants;
-    const excludeKeys = ["progress", "indicator", "step"];
+    const classes = computed(() => {
+      const value = config.value[key];
 
-    if (isVariants && !excludeKeys.includes(key)) {
-      const getCVA = cva(value);
+      if (value.variants || value.compoundVariants) {
+        return setColor(
+          cva(value)({
+            ...props,
+            color: getColor(props.color),
+          }),
+          props.color,
+        );
+      }
 
-      classes = computed(() => getCVA({ ...props }));
-    }
+      return "";
+    });
 
-    if (isVariants && excludeKeys.includes(key)) {
-      const getCVA = cva(value);
-
-      classes = computed(() =>
-        setColor(getCVA({ ...props, color: getColor(props.color) }), props.color),
-      );
-    }
+    attrs[`${key}Attrs`] = getAttrs(key, { classes });
 
     if (key === "step") {
-      const stepAttrsRaw = getAttrs("step", { classes });
+      const stepAttrs = getAttrs("step", { classes });
```

Comment:
Don't need to call same function second time, use result from variable. And fix same issue in all componenents where you did in the same way.

```js
const stepAttrs = attrs[`${key}Attrs`];
```